File: fitquest/lib/models/exercise.dart
// fitquest/lib/models/exercise.dart
class Exercise {
  final int? id;
  final String name;
  final String category;
  final String description;
  final String imageUrl;
  final bool isSynced;
  final DateTime createdAt;

  Exercise({
    this.id,
    required this.name,
    required this.category,
    required this.description,
    required this.imageUrl,
    this.isSynced = true,
    DateTime? createdAt,
  }) : createdAt = createdAt ?? DateTime.now();

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'category': category,
      'description': description,
      'imageUrl': imageUrl,
      'isSynced': isSynced ? 1 : 0,
      'createdAt': createdAt.toIso8601String(),
    };
  }

  factory Exercise.fromMap(Map<String, dynamic> map) {
    return Exercise(
      id: map['id'],
      name: map['name'],
      category: map['category'],
      description: map['description'],
      imageUrl: map['imageUrl'],
      isSynced: map['isSynced'] == 1,
      createdAt: DateTime.parse(map['createdAt']),
    );
  }
}

File: fitquest-api/models/Exercise.js
const mongoose = require('mongoose');

const ExerciseSchema = new mongoose.Schema({
  owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    index: true,
  },
  name: {
    type: String,
    required: true,
    trim: true,
  },
  category: {
    type: String,
    required: true,
    trim: true,
  },
  description: {
    type: String,
    default: '',
    trim: true,
  },
  imageUrl: {
    type: String,
    default: '',
    trim: true,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
  updatedAt: {
    type: Date,
    default: Date.now,
  },
});

ExerciseSchema.index({ owner: 1, name: 1, category: 1 });

module.exports = mongoose.model('Exercise', ExerciseSchema);
